### DISCUSSION TITLE
LSPosed/lsparself will be public?

### DISCUSSION THREAD

**Discussion by yashikada** [LSPosed/LSPlant#67] [Q&A]
Im testing hook on Android 15 DP1 and I have an issue.
I saw that repository went from SandHook to lsparser to get the symbols from the libart.so library, only that lsparser is not public, will it be public or should I write an elf parser myself?

**Comment by mywalkb** [comment]
I'm reversing ghidra source which can find symbols from a stripped library.
I don't think lsparself will ever be public.

**Comment by mywalkb** [comment]
[Here](https://github.com/mywalkb/LSPosed_mod/blob/android15/core/src/main/jni/src/elf_util.cpp) you can find a source that parses the gnudebugdata header, unpacks it and parses it again, so the symtab header is available again and the stripped libart.so library can work with lsplant.
I don't understand why not to make that repository public, this behavior gives the idiots the upper hand.

Idiots should be ignored and continued as if nothing had happened.
A lion does not care about the judgment of a goat.

**Comment by p2kyc0ng** [comment]
there is a lot of elf parse library on android, why the lsparself is not opensource?
